### PR TITLE
fix: remove dependency on request-promise

### LINKED ISSUE #744: [Bug] request package is deprecated
## Describe the bug
[Request ](https://www.npmjs.com/package/request) package is deprecated. We can consider using [node-fetch](https://www.npmjs.com/package/node-fetch) package or other similar packages instead.

### PR DESCRIPTION
**Pull request by pownkel**
#### Description of changes

Remove dependency on `request` and `request-promise`, which are deprecated, and replace them with [got](https://www.npmjs.com/package/got). This PR also moves some common logic and types related to web requests into the `common` module.

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #744
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/azure-services/src/app-insights-api-client/application-insights-client.ts [hunk 1]
```diff
@@ -2,37 +2,36 @@
...
+    private readonly defaultOptions: ExtendOptions = {
+        responseType: 'json',
     };
 
-    constructor(private readonly appId: string, private readonly apiKey: string, httpRequest: any = requestPromise) {
-        this.defaultRequestObject = httpRequest.defaults({
+    constructor(
+        private readonly appId: string,
+        private readonly apiKey: string,
+        gotRequest: any = got,
```
Comment (lamaks) [inline packages/azure-services/src/app-insights-api-client/application-insights-client.ts:26]: may be `request: got = got,`

File: packages/azure-services/src/app-insights-api-client/application-insights-client.ts [hunk 2]
```diff
@@ -2,37 +2,36 @@
...
+        getAgents: () => Agents = getForeverAgents,
```
Comment (lamaks) [inline packages/azure-services/src/app-insights-api-client/application-insights-client.ts:27]: may be `getAgentsFn: () ...`

File: packages/web-api-client/src/a11y-service-client.ts [hunk 3]
```diff
@@ -37,21 +29,23 @@ export class A11yServiceClient {
         private readonly logger: Logger,
         private readonly apiVersion = '1.0',
         private readonly throwOnRequestFailure: boolean = false,
-        private readonly httpRequest: any = requestPromise,
+        private readonly gotRequest: Got = got,
```
Comment (lamaks) [inline packages/web-api-client/src/a11y-service-client.ts:32]: may be `request: got = got,`

File: packages/web-api-client/src/a11y-service-client.ts [hunk 4]
```diff
@@ -84,7 +78,7 @@ export class A11yServiceClient {
         const requestUrl: string = `${this.requestBaseUrl}/scans/${scanId}/reports/${reportId}`;
 
         return (await this.retryHelper.executeWithRetries(
-            async () => (await this.signRequest()).get(requestUrl),
+            async () => (await this.signRequest()).get(requestUrl, { responseType: 'text' }),
```
Comment (lamaks) [inline packages/web-api-client/src/a11y-service-client.ts:81]: text/html is probably more accurate

Comment (pownkel) [reply packages/web-api-client/src/a11y-service-client.ts:81]: This is to tell got how to parse the response body, it only accepts 'json', 'text', or 'buffer.'

**Review by lamaks — APPROVED** [review]

**Pull request merged**